packages/tooling/fast-tooling/src/message-system/webworker.spec.md

```diff
@@ -0,0 +1,75 @@
+# Abstract
+
+This spec outlines the functionality of the message system web worker which will store & enact data updates, store & enact navigation updates and store & enact history updates.
+
+## Overview
+
+The [web worker](./webworker.ts) is used as an entrypoint for the generated message system web worker. The main logic for data, navigation and history manipulation is contained in the [message system utilities](./message-system.utilities.ts). A single minified file is created and bundled with the package as a web worker a consumer can reference. Using the `MessageSystem` defined in `message-system.ts`, this web worker will be passed during instantiation so that the `MessageSystem` can control and perform the data, navigation and history updates.
+
+## Data
+
+<-- TODO: document the data interactions see #3822. -->
+TBD
+
+## Navigation
+
+<-- TODO: document the navigation interactions see #3823. -->
+TBD
+
+## History
+
+### Stored history
+
+A history of different actions (data, navigation) will be stored as a local variable.
+
+Example of a history with 3 items:
+
+```ts
+[
+    {
+        type: "initialize",
+        dataDictionary: [
```

Comment:
Is the data dictionary associated with the action a complete record of the state of the app or design? Or is it a diff from the previous dictionary state?

Reply:
It's essentially a diff, because it will just be the previous outgoing message. Do you think it should be a complete record?

Reply:
No. In fact I had some concerns if it was a complete record. We're good :)